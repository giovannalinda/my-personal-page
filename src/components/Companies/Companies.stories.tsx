import { Meta, Story } from '@storybook/react'
import { Companies } from './Companies'

export default {
  title: 'Components/Companies',
  component: Companies,
} as Meta

export const Default: Story = () => <Companies />
