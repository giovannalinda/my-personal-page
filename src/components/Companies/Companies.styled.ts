import { motion } from 'framer-motion'
import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  margin-bottom: 30rem;
`

export const Title = styled.h1`
  font-size: ${theme.font.sizes.subtitle};
  font-weight: ${theme.font.weigths.regular};
  padding-bottom: ${theme.space[7]};

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    padding-left: ${theme.space[7]};
  }
`

export const List = styled.ul`
  display: flex;
  align-items: center;
  justify-content: center;
`

export const ListItem = styled(motion.li)`
  background: ${theme.colors.gray900};
  padding: 8rem ${theme.space[7]};
  width: 100%;
  transition: ${theme.transitions.slow};
  border: 1px solid ${theme.colors.gray900};

  &:hover {
    border-color: ${theme.colors.white};
  }

  & + & {
    margin-left: ${theme.space[4]};
  }

  h2 {
    padding-bottom: ${theme.space[3]};
    font-size: ${theme.font.sizes.paragraph};
    font-weight: ${theme.font.weigths.medium};
  }
`

export const Paragraph = styled.p`
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.gray400};
`
