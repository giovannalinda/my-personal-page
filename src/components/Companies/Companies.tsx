import { InView } from 'react-intersection-observer'

import * as S from './Companies.styled'

const companies = [
  {
    title: 'Front-end Developer',
    paragraph: 'Hypesoft • 6 meses',
  },
  {
    title: 'Front-end Developer',
    paragraph: '42 Dias Tecnologia • 5 meses',
  },
]

const cardVariant = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
  },
}

export function Companies() {
  return (
    <InView triggerOnce>
      {({ inView, ref }) => (
        <S.Container ref={ref}>
          <S.Title>Carreira</S.Title>

          {inView && (
            <S.List>
              {companies.map((company, index) => (
                <S.ListItem
                  key={company.paragraph}
                  aria-label={company.title}
                  initial='hidden'
                  animate='visible'
                  variants={cardVariant}
                  transition={{
                    duration: 0.5,
                    delay: index / companies.length,
                  }}
                >
                  <h2>{company.title}</h2>
                  <S.Paragraph>{company.paragraph}</S.Paragraph>
                </S.ListItem>
              ))}
            </S.List>
          )}
        </S.Container>
      )}
    </InView>
  )
}
