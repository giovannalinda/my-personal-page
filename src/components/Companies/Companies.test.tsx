import { render } from '~/__helpers__/app-tests'
import { Companies } from './Companies'

describe('Companies', () => {
  it('matches snapshot', () => {
    const { container } = render(<Companies />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
