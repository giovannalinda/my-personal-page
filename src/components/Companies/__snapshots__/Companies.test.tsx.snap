// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies matches snapshot 1`] = `
<section
  class="sc-bczRLJ jhDWfm"
>
  <h1
    class="sc-gsnTZi dSgLBd"
  >
    Carreira
  </h1>
  <ul
    class="sc-dkzDqf jStVNb"
  >
    <li
      class="sc-hKMtZM jkumoH"
    >
      <h2>
        Front-end Developer
      </h2>
      <span>
        Hypesoft • 6 meses
      </span>
    </li>
    <li
      class="sc-hKMtZM jkumoH"
    >
      <h2>
        Front-end Developer
      </h2>
      <span>
        42 Dias Tecnologia • 5 meses
      </span>
    </li>
  </ul>
</section>
`;
